src/test/java/pl/wrapper/parking/facade/domain/ParkingControllerFreeSpotsTest.java

```diff
@@ -0,0 +1,202 @@
+package pl.wrapper.parking.facade.domain;
+
+import static org.hamcrest.CoreMatchers.anything;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import java.time.LocalTime;
+import java.util.List;
+import org.json.JSONArray;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import pl.wrapper.parking.facade.ParkingService;
+import pl.wrapper.parking.infrastructure.error.ParkingError;
+import pl.wrapper.parking.infrastructure.error.Result;
+import pl.wrapper.parking.pwrResponseHandler.dto.ParkingResponse;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class ParkingControllerFreeSpotsTest {
```

Comment:
We shouldn't start a full test context here. Also move the unit tests to the ParkingControllerTests, there is no need to create a separate class for them